packages/ecommerce-ui/components/FormComponents/TextFieldElement.tsx

```diff
@@ -49,9 +50,10 @@ export function TextFieldElement<TFieldValues extends FieldValues>({
           name={name}
           value={value ?? ''}
           onChange={(ev) => {
-            onChange(
-              type === 'number' && ev.target.value ? Number(ev.target.value) : ev.target.value,
-            )
+            const fieldValue =
+              type === 'number' && ev.target.value ? Number(ev.target.value) : ev.target.value
+            const transformValue = transform ? transform(fieldValue) : fieldValue
```

Comment:
`transform` prop isn't used, so can be removed?

Reply:
transform prop is used by the BundleOptionValue component